### ISSUE TITLE
[master] Can't remove alias and service

### ISSUE THREAD

**Issue by dgoradia** [HyperBrain/serverless-aws-alias#116]
I've got a single alias: `qa` and when I try to remove it I get:
```
Serverless Error ---------------------------------------

Remove the other deployed aliases before removing the service:

Get Support --------------------------------------------
...
```

When running it with `SLS_DEBUG=*`:
```
...
Serverless: Invoke alias:remove
Serverless: WARNING: Inappropriate call of provider.request()
Serverless: WARNING: Inappropriate call of provider.request()
Serverless: WARNING: Inappropriate call of provider.request()

  Serverless Error ---------------------------------------

  Remove the other deployed aliases before removing the service:

  Stack Trace --------------------------------------------

ServerlessError: Remove the other deployed aliases before removing the service:
    at AwsAlias.removeAlias (/Users/dgoradia/D/go/src/p.com/node_modules/serverless-aws-alias/lib/removeAlias.js:229:11)
From previous event:
    at PluginManager.invoke (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:372:22)
    at PluginManager.run (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:403:17)
    at variables.populateService.then (/usr/local/lib/node_modules/serverless/lib/Serverless.js:102:33)
    at runCallback (timers.js:756:18)
    at tryOnImmediate (timers.js:717:5)
    at processImmediate [as _immediateCallback] (timers.js:697:5)
From previous event:
    at Serverless.run (/usr/local/lib/node_modules/serverless/lib/Serverless.js:89:74)
    at serverless.init.then (/usr/local/lib/node_modules/serverless/bin/serverless:42:50)
    at <anonymous>
```

**Comment by HyperBrain** [comment]
Hi @dgoradia , your stage is also "qa"? If not, you have 2 aliases one with the stage name and one with "qa". You should remove the aliase with `serverless alias remove --alias=YYYY --stage=XXXX` first.

`serverless remove --stage=XXXX` only works if there are no other aliases deployed.

**Comment by dgoradia** [comment]
Yes, my stage name, coincidentally is "qa" which I need to specify the aws profile sls should use:
```
...
provider:
  name: aws
  runtime: go1.x
  stage: ${opt:stage, self:custom.defaultStage}
  region: us-west-2
  profile: ${self:custom.profiles.${self:provider.stage}}
...
```

I've tried removing it like that:
```
$ sls alias remove --alias=qa --stage=qa

  Serverless Error ---------------------------------------

  Remove the other deployed aliases before removing the service:
```

**Comment by HyperBrain** [comment]
Can you try to execute `serverless info --verbose`? It should emit a list of deployed aliases at the very end of the output.

**Comment by dgoradia** [comment]
```
$ sls --stage=qa info --verbose
Service Information
service: partner-api
stage: qa
region: us-west-2
stack: partner-api-qa
api keys:
  None
endpoints:
  GET - https://t7xlc6jy14.execute-api.us-west-2.amazonaws.com/qa/keys
  POST - https://t7xlc6jy14.execute-api.us-west-2.amazonaws.com/qa/keys
  PUT - https://t7xlc6jy14.execute-api.us-west-2.amazonaws.com/qa/keys
  DELETE - https://t7xlc6jy14.execute-api.us-west-2.amazonaws.com/qa/keys/{id}
  POST - https://t7xlc6jy14.execute-api.us-west-2.amazonaws.com/qa/tax
  POST - https://t7xlc6jy14.execute-api.us-west-2.amazonaws.com/qa/age
functions:
  keys_get: partner-api-qa-keys_get
  keys_post: partner-api-qa-keys_post
  keys_put: partner-api-qa-keys_put
  keys_delete: partner-api-qa-keys_delete
  tax_post: partner-api-qa-tax_post
  age_post: partner-api-qa-age_post

Stack Outputs
ApiGatewayRestApiRootResource: thep5kacwi
KeysUnderscorepostLambdaFunctionArn: arn:aws:lambda:us-west-2:782555603950:partner-api-qa-keys_post
KeysUnderscoredeleteLambdaFunctionArn: arn:aws:lambda:us-west-2:782555603950:partner-api-qa-keys_delete
ApiGatewayRestApi: t7xlc6jy14
KeysUnderscoregetLambdaFunctionArn: arn:aws:lambda:us-west-2:782555603950:partner-api-qa-keys_get
AgeUnderscorepostLambdaFunctionArn: arn:aws:lambda:us-west-2:782555603950:partner-api-qa-age_post
TaxUnderscorepostLambdaFunctionArn: arn:aws:lambda:us-west-2:782555603950:partner-api-qa-tax_post
ServiceEndpoint: https://t7xlc6jy14.execute-api.us-west-2.amazonaws.com/qa
ServerlessDeploymentBucketName: partner-api-qa-serverlessdeploymentbucket-am9lf9nm5j1c
ServerlessAliasReference: REFERENCE
KeysUnderscoreputLambdaFunctionArn: arn:aws:lambda:us-west-2:782555603950:partner-api-qa-keys_put

aliases:
  qa
    Functions:
      partner-api-qa-tax_post -> 5
      partner-api-qa-keys_delete -> 5
      partner-api-qa-keys_get -> 5
      partner-api-qa-keys_post -> 5
      partner-api-qa-keys_put -> 5
      partner-api-qa-age_post -> 5
```

**Comment by dgoradia** [comment]
In CF I have 2 stacks: `partner-api-qa` and `partner-api-qa-qa`.

**Comment by HyperBrain** [comment]
Yes, two stacks is ok - one is the alias stack and one is the base stack.

However, you only have one stage and one alias, so you'd remove the whole service. Is that what you wanted to do? Then `serverless remove --stage=qa` should work and remove the service stage completely.

**Comment by dgoradia** [comment]
Yes that's what I want to do and that's where I'm getting the error:
```
$ serverless remove --stage=qa

  Serverless Error ---------------------------------------

  Use "serverless alias remove --alias=qa" to remove the service.
```

```
$ serverless alias remove --alias=qa

  Serverless Error ---------------------------------------

  Remove the other deployed aliases before removing the service:
```

**Comment by dgoradia** [comment]
Here's my serverless.yml file. Let me know what I can do on my end to help reproduce this.

```
service: partner-api
frameworkVersion: "=1.26.1"

custom:
  simulate:
    services:
      file: docker-compose.yml
      projectName: partnerapi
  local:
    dynamodb:
      port: 8000
  defaultStage: developer
  profiles: // These refer to aws profiles
    developer: local
    qa: qa
    staging: prod
    production: prod

provider:
  name: aws
  runtime: go1.x
  stage: ${opt:stage, self:custom.defaultStage}
  region: us-west-2
  profile: ${self:custom.profiles.${self:provider.stage}}

package:
 exclude:
   - ./**
 include:
   - ./bin/**
   - ./config/**

functions:
  keys_get:
    handler: bin/keys_get
    events:
      - http:
          method: GET
          path: keys
          request:
            parameters:
              querystrings:
                id: false

plugins:
  - serverless-sam
  - serverless-aws-alias
  - serverless-plugin-simulate
```

**Comment by dgoradia** [comment]
@HyperBrain I just realized I was using the master branch of this to get support for domain manager which is due in 1.7.0 but after reverting back to 1.6.1, `alias remove` works as intended so it's broken in master.

**Comment by HyperBrain** [comment]
@dgoradia Good to know. This has to be fixed before the next release. Thank you.

**Comment by aleksdikanski** [comment]
Looks like I used the wrong lodash function for filtering the master alias. Should be fixed by #117. I reproduced the problem on my test project and the fix should work. @dgoradia could your verify that this works by using my branch?

**Comment by dgoradia** [comment]
@aleksdikanski verified that it's fixed in #117 
Thank you!

**Comment by ashish-akshantal** [comment]
> Hi @dgoradia , your stage is also "qa"? If not, you have 2 aliases one with the stage name and one with "qa". You should remove the aliase with `serverless alias remove --alias=YYYY --stage=XXXX` first.
> 
> `serverless remove --stage=XXXX` only works if there are no other aliases deployed.
> 
> I think I implemented `serverless alias list --stage=XXXX` too, to show actually deployed aliases.

Hi @HyperBrain 
sls alias list --stage=dev does not work, neither does serverless alias remove --alias=v102.
I get theese errors while running the above command 
Serverless command "alias remove" not found
Serverless command "alias list" not found
